### PR TITLE
Call K2PgExecuteInsert instead of heap insert for K2 relation

### PR DESCRIPTION
**Pull request by ahsank**

### CODE REVIEW

File: src/gausskernel/runtime/executor/nodeModifyTable.cpp [hunk 1]
```diff
@@ -992,7 +992,12 @@ TupleTableSlot* ExecInsertT(ModifyTableState* state, TupleTableSlot* slot, Tuple
                                 searchHBucketFakeRelation(estate->esfRelations, estate->es_query_cxt,
                                     result_relation_desc, bucket_id, target_rel);
                             }
-                            new_id = tableam_tuple_insert(target_rel, tuple, estate->es_output_cid, 0, NULL);
+                            
+                            if (IsK2PgRelation(target_rel)) {
```
Comment (johnfangAFW) [inline src/gausskernel/runtime/executor/nodeModifyTable.cpp]: Could do a more generic update at heap_insert method in heapam.cpp 

https://github.com/futurewei-cloud/chogori-opengauss/commit/6db1db72b47a294b48cdedf6d37fd88110f51fd0#diff-07e6dc90be18e26d222004fc6c439178c7c68cdd8e4e43690ca04af501e35021

File: src/gausskernel/storage/access/heap/heapam.cpp [hunk 2]
```diff
@@ -2789,12 +2789,12 @@ Oid heap_insert(Relation relation, HeapTuple tup, CommandId cid, int options, Bu
         Assert(TUPLE_IS_HEAP_TUPLE(tup));
     }
 
-    if (IsK2PgRelation(relation))
-    {
-	    ereport(ERROR,
+    if (IsK2PgRelation(relation)) {
+        ereport(WARNING,
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/heap/heapam.cpp:2793]: please change ereport to elog or remove the ereport line. thanks

File: src/gausskernel/runtime/executor/nodeModifyTable.cpp (hunk 1 above)
Comment (ahsank) [reply src/gausskernel/runtime/executor/nodeModifyTable.cpp]: I forgot to check the link so just moved the code. It looks same except the warning. Not sure if we should keep the warning.
```
PostgreSQL stand-alone backend 9.2.4
backend> INSERT INTO pg_depend SELECT 0,0,0, tableoid,oid,0, 'p' FROM pg_ts_config;
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.292 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.354 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.362 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
errstart level: 19, heapam.cpp 2796 heap_insert
2022-11-15 19:15:20.370 [unknown] [unknown] localhost 140737353960768 0[0:0#0]  [BACKEND] WARNING:  Using K2PG heap insert in heap_insert
backend>
```

Comment (johnfangAFW) [reply src/gausskernel/runtime/executor/nodeModifyTable.cpp]: let us remove the warning or use elog as debug level log

**Comment by johnfangAFW** [comment]
Also, please change your PR against mainline. Bugbash branch already got my fix there.

**Pull request closed without merging**